lib/logstash/filters/grok.rb

```diff
@@ -390,7 +390,7 @@ def patterns_files_from_paths(paths, glob)
 
         Dir.glob(path).each do |file|
           @logger.trace("Grok loading patterns from file", :path => file)
-          patternfiles << file
+          patternfiles << file unless File.directory?(file)
```

Comment:
Let's log when we are skipping things. Perhaps at `debug` level, `logger.debug("Skipping path because it is a directory", :path => file)`